tests/integration/cartography/intel/github/test_repos.py

```diff
@@ -186,3 +186,44 @@ def test_repository_to_collaborators(neo4j_session):
     actual_nodes = {n['collab_count'] for n in nodes}
     expected_nodes = {5}
     assert actual_nodes == expected_nodes
+
+
+def test_pinned_python_library_to_repo(neo4j_session):
+    """
+    Ensure that repositories are connected to pinned Python libraries.
+    Create the path (:RepoA)-[:REQUIRES{specifier:"0.1.0"}]->(:PythonLibrary{'Cartography'})<-[:REQUIRES]-(:RepoB),
+    and verify that exactly 1 repo is connected to the PythonLibrary with a specifier (RepoA).
+    """
+    _ensure_local_neo4j_has_test_data(neo4j_session)
+
+    # Note: don't query for relationship attributes in code that needs to be fast.
+    query = """
+    MATCH (repo:GitHubRepository)-[r:REQUIRES]->(lib:PythonLibrary{id:'cartography'})
+    WHERE r.specifier = "0.1.0"
+    RETURN count(repo) as repo_count
+    """
+    nodes = neo4j_session.run(query)
+    actual_nodes = {n['repo_count'] for n in nodes}
+    expected_nodes = {1}
+    assert actual_nodes == expected_nodes
+
+
+def test_upinned_python_library_to_repo(neo4j_session):
```

Comment:
Would be good to have a test case for a URL-based dep too, to make sure we ignore it OK (doesn't make it into the graph) without erroring out. (packaging will throw an InvalidRequirement and we'll skip it).